cat-shark-carousel/src/components/pagination.tsx

```diff
@@ -0,0 +1,73 @@
+import React from 'react';
+
+interface IProps {
+    array: any[];
+    activeItem: any;
+    pages?: number;
+    onClick: (index: number) => void;
+}
+
+const Pagination = ({ array, activeItem, pages = 5, onClick }: IProps) => {
+    let pageOffset = pages;
+    if (pages % 2 === 0) {
+        pageOffset += 1
+    }
+
+    if (array.length === 0) {
+        return null;
+    }
+
+    let offset = (pageOffset - 1) / 2;
+    let firstDots = true;
+    let lastDots = true;
+    let snippet = [];
+    // Simple algorithm tries to keep the activeIndex in the middle of the
+    // pagination.
+    let activeIndex = array.findIndex((value) => value === activeItem);
+    let middleItem = activeIndex;
+    let startingIndex = middleItem - offset - 1;
+    if (pages < array.length) {
+        if (startingIndex < 0) {
+            middleItem = offset;
+            firstDots = false
+        }
+        if (activeIndex + offset >= array.length - 1) {
+            middleItem = array.length - 1 - offset;
+            lastDots = false;
+        }
+        startingIndex = middleItem - offset;
+        snippet = array.slice(startingIndex, startingIndex + pages);
+    } else {
+        snippet = array;
+        startingIndex = 0;
+        firstDots = false
+        lastDots = false;
+    }
```

Comment:
Why did you decide not to add this in a Hook?

Reply:
I didn’t add this as a hook because it was one of the last things I was completing. It was already late and I was just adding this as one of the final pieces of the frontend. It never even occurred to me to make this into a hook as I didn’t think I’d reuse it anywhere else in the application, thus containing the pagination logic in the component was fine. However, now that you mention it, this logic would be better suited as a hook and I wish I did it that way.